### PR TITLE
NIFI-11938: Created ConsumeSlack and ListenSlack Processors to consume message events and commands from Slack

### PR DESCRIPTION
**Pull request by markap14**
# Summary

[NIFI-00000](https://issues.apache.org/jira/browse/NIFI-00000)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/consume/ConsumeChannel.java [hunk 1]
```diff
@@ -0,0 +1,861 @@
...
+        String minTsValue = stateMap.get(stateKeys.REPLY_MIN_TS);
+        if (minTsValue == null) {
+            minTsValue = latestTs;
+        }
+        final String maxTsValue = stateMap.get(stateKeys.REPLY_MAX_TS);
+        final SlackTimestamp minTs = new SlackTimestamp(minTsValue);
+        final SlackTimestamp maxTs = maxTsValue == null ? new SlackTimestamp() : new SlackTimestamp(maxTsValue);
+        final SlackTimestamp maxParentTs = new SlackTimestamp(latestTs);
+
+        // TODO: Test this heavily
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/consume/ConsumeChannel.java:224]: Should it be removed?

**Comment by joewitt** [comment]
Build is good but we need L&N.

For the slack lib added it is at least: https://github.com/SlackAPI/java-slack-sdk/blob/main/LICENSE

That will need to be in nifi-assembly/LICENSE as it is MIT.

Looking at transitive deps and will provide more

**Comment by joewitt** [comment]
I also think these libs pull in some rather dated libraries.  We should likely override them to their newer counterparts as the CVE nature of these is often considerable.  Looking in depth now

**Comment by exceptionfactory** [comment]
> I also think these libs pull in some rather dated libraries. We should likely override them to their newer counterparts as the CVE nature of these is often considerable. Looking in depth now

Thanks for looking into this @joewitt, we should definitely override to the latest versions of items whenever we can, especially when introducing new libraries.

**Comment by joewitt** [comment]
should bump to slack lib 1.32.1

**Pull request closed without merging**